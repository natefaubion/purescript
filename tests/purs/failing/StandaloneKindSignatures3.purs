-- @shouldFailWith KindsDoNotUnify
module Main where

type Fst :: forall k. k -> k -> k
type Fst a b = a

type F = Fst Int "foo"
