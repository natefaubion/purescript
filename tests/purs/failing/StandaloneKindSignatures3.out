Error found:
in module [33mMain[0m
at tests/purs/failing/StandaloneKindSignatures3.purs:7:18 - 7:23 (line 7, column 18 - line 7, column 23)

  Could not match kind
  [33m        [0m
  [33m  Symbol[0m
  [33m        [0m
  with kind
  [33m      [0m
  [33m  Type[0m
  [33m      [0m

while checking that type [33m"foo"[0m
  has kind [33mType[0m
while inferring the kind of [33mFst Int "foo"[0m
in type synonym [33mF[0m

See https://github.com/purescript/documentation/blob/master/errors/KindsDoNotUnify.md for more information,
or to contribute content related to this error.

